specification/monitor/data-plane/ingestion/stable/2023-01-01/DataCollectionRules.json

```diff
@@ -1,136 +1,215 @@
       "post": {
-        "tags": [
-          "Data Collection Rules"
-        ],
-        "summary": "Ingestion API used to directly ingest data using Data Collection Rules",
-        "description": "See error response code and error response message for more detail.",
         "operationId": "Upload",
-        "consumes": [
-          "application/json"
-        ],
-        "produces": [
-          "application/json"
-        ],
+        "summary": "Ingestion API used to directly ingest data using Data Collection Rules.",
+        "description": "Ingestion API used to directly ingest data using Data Collection Rules.",
         "parameters": [
           {
-            "in": "path",
-            "name": "ruleId",
-            "description": "The immutable Id of the Data Collection Rule resource.",
-            "required": true,
-            "type": "string"
+            "$ref": "#/parameters/Azure.Core.Foundations.ApiVersionParameter"
           },
           {
+            "name": "ruleId",
             "in": "path",
-            "name": "stream",
-            "description": "The streamDeclaration name as defined in the Data Collection Rule.",
+            "description": "The immutable ID of the Data Collection Rule resource.",
             "required": true,
             "type": "string"
           },
           {
-            "in": "query",
-            "name": "api-version",
+            "name": "stream",
+            "in": "path",
+            "description": "The streamDeclaration name as defined in the Data Collection Rule.",
             "required": true,
             "type": "string",
-            "description": "The API version to use for this operation."
+            "x-ms-client-name": "streamName"
           },
           {
-            "in": "header",
             "name": "Content-Encoding",
-            "description": "gzip",
-            "type": "string"
+            "in": "header",
+            "description": "The content encoding of the request body which is always 'gzip'.",
+            "required": false,
+            "type": "string",
+            "x-ms-client-name": "contentEncoding"
           },
           {
-            "in": "header",
-            "name": "x-ms-client-request-id",
-            "description": "Client request Id",
-            "type": "string"
+            "$ref": "#/parameters/Azure.Core.ClientRequestIdHeader"
           },
           {
-            "in": "body",
             "name": "body",
-            "description": "An array of objects matching the schema defined by the provided stream.",
+            "in": "body",
+            "description": "The array of objects matching the schema defined by the provided stream.",
             "required": true,
             "schema": {
               "type": "array",
               "items": {
-                "type": "object"
-              }
+                "type": "object",
+                "additionalProperties": {}
```

Comment:
I have to double check this but I believe this will be generated as a different type..

Reply:
Ah, yea. I do see that listed as a breaking change in the breaking change check. Do you know if this is from the generation from the `body: Record<unknown>[];` typing?

Reply:
From what I can tell, `"additionalProperties": {}` indicates that each object in the array can have any number of properties with any type which seems right.

Reply:
Ok I checked with Johan and I think this representation is equivalent. However, I'm not sure if code generators will treat this the same as before :/ Have you tried generating from this spec? And have you noticed any differences related to this implementation?

Reply:
Yea, for python, generation seems good (one non-blocking [issue](https://github.com/Azure/autorest.python/issues/2827) related to typing, however).